src/UI/Seller/src/app/product-management/components/products-form/products-form.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Product } from '@ordercloud/angular-sdk';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { AppFormErrorService, AppIdValidator } from '@app-seller/shared';
+
+@Component({
+  selector: 'products-form',
+  templateUrl: './products-form.component.html',
+  styleUrls: ['./products-form.component.scss'],
+})
+export class ProductsFormComponent implements OnInit {
+  private _existingProduct: Product = {};
+  @Input() btnText: string;
+  @Output() formSubmitted = new EventEmitter();
+  productForm: FormGroup;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private formErrorService: AppFormErrorService
+  ) {}
+
+  ngOnInit() {
+    this.setForm();
+  }
+
+  @Input()
+  set existingProduct(product: Product) {
+    this._existingProduct = product || {};
+    this.setForm();
+  }
```

Comment:
What does this do? Anytime the product changes, it creates a new form? Is that right? So is it kind of like ngOnChanges but not?

Reply:
Yeah, like ngOnChanges but just when the product input changes. This is a good catch, it should probably use form.setValue() instead of re-creating the formGroup.